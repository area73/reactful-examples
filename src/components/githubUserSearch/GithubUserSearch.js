import React from 'react';

import CardList from './CardList.js';
import GitHubForm from './GitHubForm.js';
import Toaster from './Toaster.js';

class GithubUserSearch extends React.Component {

  state = {
    profiles: [],
    error: [],
  };

  addProfile = profileData => {
    this.setState(prevState => ({
      profiles: [...prevState.profiles, profileData],
      error:[],
    }));
  };

  removeProfile = profileId => {
    this.setState(prevState => ({
      profiles: prevState.profiles.filter(profile => profile.id !== profileId),
    }));
  };

  addError = err => {
    this.setState(() => (
      {
        error: err,
      }
    ));
  }

  render() {
    const title = this.props.initialData && this.props.initialData.appName ||'Hello World';
    return (
      <div>
        <h1>{title}</h1>
        <GitHubForm onSubmit={this.addProfile} onSubmitError={this.addError}/>
        <Toaster type="error" data={this.state.error} />
        <CardList profiles={this.state.profiles} update={this.removeProfile}/>
      </div>
    );

  }
}

export default GithubUserSearch;
