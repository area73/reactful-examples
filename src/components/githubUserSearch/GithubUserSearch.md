#User Search
<hr>
App compose of:
  * Search
  * CardList
  * Card
  * Toaster
