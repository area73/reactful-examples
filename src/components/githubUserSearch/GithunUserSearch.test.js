import React from 'react';
import GithubUserSearch  from './GithubUserSearch.js';
import renderer from 'react-test-renderer';


describe('CardList', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<GithubUserSearch initialData={{appName:'Title page'}}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
