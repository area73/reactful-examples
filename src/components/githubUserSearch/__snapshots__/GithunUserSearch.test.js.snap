// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList renders correctly 1`] = `
<div>
  <h1>
    Title page
  </h1>
  <div
    className="github-form"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="GitHub user name"
        required={true}
        type="text"
        value=""
      />
      <button>
        Add Card
      </button>
    </form>
  </div>
  <div
    className="toaster"
  >
    <span
      className="error  hide"
    />
  </div>
  <div
    className="card-list"
  />
</div>
`;
